static/js/admin.js

```diff
@@ -959,13 +959,15 @@ function _setup_page() {
         var aliases_info = $("#realm_aliases_modal").find(".aliases_info");
         var data = {
             domain: JSON.stringify($("#new_alias").val()),
+            allow_subdomains: JSON.stringify($("#allow_subdomains").prop("checked")),
         };
 
         channel.post({
             url: "/json/realm/domains",
             data: data,
             success: function () {
                 $("#new_alias").val("");
+                $("#allow_subdomains").prop("checked", false);
```

Comment:
Yeah, that's a lot clearer approach.  I'm going to try applying that, fixing a couple more small things, and then merge this.